examples/mini-app/src/components/ui/slider.tsx

```diff
@@ -1,34 +1,131 @@
 import * as SliderPrimitive from "@radix-ui/react-slider"
 import * as React from "react"
-
 import { cn } from "../../lib/utils"
+import { Tooltip, TooltipContent, TooltipProvider, TooltipTrigger } from "./tooltip"
+
+function getAdjustedPercentage(value: number, min: number, max: number): number {
+  const basePercentage = ((value - min) / (max - min)) * 100
+
```

Comment:
This calculation runs on every render and could impact performance. Consider memoizing it:
\\\